main.go

```diff
@@ -137,7 +143,19 @@ func (h *handler) listContainers(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	err = json.NewEncoder(w).Encode(containers)
+
+    filteredContainers := []docker.Container{}
+	for _, container := range containers {
```

Comment:
I am confused. I thought here you would want to pass `nameFilter` to `ListContainers`? Sorry if I wasn't clear last time. I was thinking when you have `ListContainer(filter)` then you could filter all containers by name from docker and you don't need to a regex anymore? Is that too complicated? In other words, I am suggesting remove `containerRestrictions` in favor of letting docker to the filtering. This would be more clear for users of Dozzle who are familiar with Docker. The functionality of filter would mirror `docker ps`

Reply:
I mean `containerRestrictions` as a regex. You still need a filterName or something to pass to docker